--- configure.ac~	2011-03-24 16:35:39.000000000 +0000
+++ configure.ac	2011-09-26 00:15:56.000000000 +0100
@@ -418,7 +418,7 @@
 if test "x$enable_gnome3" != "xno"
 then
     PKG_CHECK_MODULES([GNOME],
-                      [libpanelapplet-3.0],
+                      [libpanelapplet-4.0],
                       [config_gnome=yes
                        have_panelapplet3=yes
                        AC_DEFINE(HAVE_PANELAPPLET3, 1, [Support for Gnome panel applet v3])],
