--- workrave-1.9.1.orig/common/src/Makefile.am	2010-09-29 20:12:49.452967126 +0100
+++ workrave-1.9.1/common/src/Makefile.am	2010-09-29 21:14:04.680229047 +0100
@@ -38,7 +38,7 @@
 libworkrave_common_la_LIBADD = ${ldadd_platform}
 
 INCLUDES = 		-W -DWORKRAVE_PKGDATADIR="\"${pkgdatadir}\"" -DWORKRAVE_DATADIR="\"${datadir}\"" \
-			-D_XOPEN_SOURCE=600 \
+			-DWORKRAVE_HOME="\"${localstatedir}/workrave/\"" -D_XOPEN_SOURCE=600 \
 			@WR_COMMON_INCLUDES@ \
 			@GLIB_CFLAGS@ @GNOME_CFLAGS@ @DBUS_CFLAGS@ 
 
@@ -48,3 +48,6 @@
 
 locale.inc:
 			python $(top_srcdir)/common/bin/create_locale.py > locale.inc
+
+install-data-local:
+	${INSTALL} -d -m 1777 ${DESTDIR}${localstatedir}/workrave
--- workrave-1.9.1.orig/common/src/Util.cc	2009-05-28 20:04:02.000000000 +0100
+++ workrave-1.9.1/common/src/Util.cc	2010-09-29 21:18:09.803637452 +0100
@@ -86,6 +86,19 @@
       ret = "./";
 
 #if defined(PLATFORM_OS_UNIX) || defined(PLATFORM_OS_OSX)
+#ifdef WORKRAVE_HOME
+      const char *user = getenv("USER");
+
+      if (user == NULL)
+	{
+	  user = "unknown";
+	}
+
+      ret = WORKRAVE_HOME;
+      ret += user;
+      ret += "/";
+      mkdir(ret.c_str(), 0700);
+#else
       const char *home = getenv("WORKRAVE_HOME");
 
       if (home == NULL)
@@ -100,6 +113,7 @@
 
           mkdir(ret.c_str(), 0700);
         }
+#endif
 #elif defined(PLATFORM_OS_WIN32)
       void *pidl;
       char buf[MAX_PATH];
