--- openssl-1.0.2/crypto/cms/cms_kari.c.old	2015-03-25 17:52:16.779155470 +0000
+++ openssl-1.0.2/crypto/cms/cms_kari.c	2015-03-25 17:55:39.285562171 +0000
@@ -398,7 +398,11 @@
      * DES3 wrap otherwise use AES wrap similar to key size.
      */
     if (EVP_CIPHER_type(cipher) == NID_des_ede3_cbc)
+#ifdef OPENSSL_NO_DES
+	return 0;
+#else
         kekcipher = EVP_des_ede3_wrap();
+#endif
     else if (keylen <= 16)
         kekcipher = EVP_aes_128_wrap();
     else if (keylen <= 24)
