--- cmake/dtrace.cmake.old	2014-11-24 19:48:22.927600983 +0000
+++ cmake/dtrace.cmake	2014-11-24 19:50:00.280908240 +0000
@@ -117,7 +117,9 @@
       )
     ELSEIF(CMAKE_SYSTEM_NAME MATCHES "Linux")
       # dtrace on Linux runs gcc and uses flags from environment
+      SET(CC_SAVED $ENV{CC})
       SET(CFLAGS_SAVED $ENV{CFLAGS})
+      SET(ENV{CC} "${CMAKE_C_COMPILER} ${CMAKE_C_COMPILER_ARG1}")
       SET(ENV{CFLAGS} ${CMAKE_C_FLAGS})
       SET(outfile "${CMAKE_BINARY_DIR}/probes_mysql.o")
       # Systemtap object
@@ -125,6 +127,7 @@
         COMMAND ${DTRACE} -G -s ${CMAKE_SOURCE_DIR}/include/probes_mysql.d.base
         -o ${outfile}
         )
+      SET(ENV{CC} ${CC_SAVED})
       SET(ENV{CFLAGS} ${CFLAGS_SAVED})
     ENDIF()
 
