diff -Nru libsoup-2.30.1.orig/docs/reference/libsoup-2.4-docs.sgml libsoup-2.30.1/docs/reference/libsoup-2.4-docs.sgml
--- libsoup-2.30.1.orig/docs/reference/libsoup-2.4-docs.sgml	2010-04-26 21:15:37.000000000 +0100
+++ libsoup-2.30.1/docs/reference/libsoup-2.4-docs.sgml	2010-06-14 18:11:58.000000000 +0100
@@ -52,11 +52,13 @@
     <xi:include href="xml/soup-value-utils.xml"/>
   </chapter>
 
+  <!--
   <chapter>
     <title>GNOME integration</title>
     <xi:include href="xml/soup-gnome-features.xml"/>
     <xi:include href="xml/soup-cookie-jar-sqlite.xml"/>
   </chapter>
+  -->
 
   <chapter>
     <title>Low-level Networking API</title>
diff -Nru libsoup-2.30.1.orig/docs/reference/Makefile.am libsoup-2.30.1/docs/reference/Makefile.am
--- libsoup-2.30.1.orig/docs/reference/Makefile.am	2010-06-14 18:10:50.000000000 +0100
+++ libsoup-2.30.1/docs/reference/Makefile.am	2010-06-14 18:13:08.000000000 +0100
@@ -36,7 +36,8 @@
 	soup-dns.h soup-auth-manager.h soup-auth-manager-ntlm.h \
 	soup-message-queue.h soup-path-map.h soup-ssl.h \
 	soup-proxy-resolver.h soup-proxy-resolver-gnome.h \
-	soup-proxy-resolver-static.h
+	soup-proxy-resolver-static.h soup-cookie-jar-sqlite.h \
+	soup-gnome.h soup-gnome-features.h
 
 # Images to copy into HTML directory.
 HTML_IMAGES = 
@@ -62,7 +63,7 @@
 	$(GNUTLS_CFLAGS)		\
 	$(LIBPROXY_LIBS)
 
-GTKDOC_LIBS = $(top_builddir)/libsoup/libsoup-gnome-2.4.la
+GTKDOC_LIBS = $(top_builddir)/libsoup/libsoup-2.4.la
 
 # include common portion ...
 include $(top_srcdir)/gtk-doc.make
