--- gcc/cp/typeck2.c.ORIG	2015-01-23 17:30:00.000000000 +0100
+++ gcc/cp/typeck2.c	2015-05-06 17:28:15.423147717 +0200
@@ -957,9 +957,16 @@
 	    }
 	}
       else if (complain & tf_error)
-	error_at (EXPR_LOC_OR_LOC (init, input_location),
-		  "narrowing conversion of %qE from %qT to %qT inside { }",
-		  init, ftype, type);
+       {
+         int savederrorcount = errorcount;
+         global_dc->pedantic_errors = 1;
+         pedwarn (EXPR_LOC_OR_LOC (init, input_location), OPT_Wnarrowing,
+                  "narrowing conversion of %qE from %qT to %qT "
+                  "inside { }", init, ftype, type);
+         if (errorcount == savederrorcount)
+           ok = true;
+         global_dc->pedantic_errors = flag_pedantic_errors;
+       }
     }
 
   return cxx_dialect == cxx98 || ok; 
