diff -ru ekiga-3.2.7_orig/lib/engine/account/bank-impl.h ekiga-3.2.7/lib/engine/account/bank-impl.h
--- ekiga-3.2.7_orig/lib/engine/account/bank-impl.h	2012-08-08 19:17:09.769376169 +0200
+++ ekiga-3.2.7/lib/engine/account/bank-impl.h	2012-08-08 19:20:15.910049088 +0200
@@ -212,7 +212,7 @@
 void
 Ekiga::BankImpl<AccountType>::add_account (gmref_ptr<AccountType> account)
 {
-  add_object (account);
+  this->add_object (account);
 
   account->questions.add_handler (questions.make_slot ());
   account->registration_event.connect (sigc::bind (sigc::mem_fun (this, &Ekiga::BankImpl<AccountType>::on_registration_event), account));
diff -ru ekiga-3.2.7_orig/lib/engine/addressbook/book-impl.h ekiga-3.2.7/lib/engine/addressbook/book-impl.h
--- ekiga-3.2.7_orig/lib/engine/addressbook/book-impl.h	2012-08-08 19:17:09.787375945 +0200
+++ ekiga-3.2.7/lib/engine/addressbook/book-impl.h	2012-08-08 19:18:31.473354729 +0200
@@ -204,7 +204,7 @@
 Ekiga::BookImpl<ContactType>::add_contact (gmref_ptr<ContactType> contact)
 {
   contact->questions.add_handler (questions.make_slot ());
-  add_object (contact);
+  this->add_object (contact);
 }
 
 
diff -ru ekiga-3.2.7_orig/lib/engine/addressbook/source-impl.h ekiga-3.2.7/lib/engine/addressbook/source-impl.h
--- ekiga-3.2.7_orig/lib/engine/addressbook/source-impl.h	2012-08-08 19:17:09.787375945 +0200
+++ ekiga-3.2.7/lib/engine/addressbook/source-impl.h	2012-08-08 19:17:47.654902533 +0200
@@ -206,7 +206,7 @@
 void
 Ekiga::SourceImpl<BookType>::add_book (gmref_ptr<BookType> book)
 {
-  add_object (book);
+  this->add_object (book);
 
   add_connection (book, book->contact_added.connect (sigc::bind<0> (contact_added.make_slot (), book)));
 
diff -ru ekiga-3.2.7_orig/lib/engine/presence/cluster-impl.h ekiga-3.2.7/lib/engine/presence/cluster-impl.h
--- ekiga-3.2.7_orig/lib/engine/presence/cluster-impl.h	2012-08-08 19:17:09.767376195 +0200
+++ ekiga-3.2.7/lib/engine/presence/cluster-impl.h	2012-08-08 19:19:12.518841588 +0200
@@ -147,7 +147,7 @@
 
   add_connection (heap, heap->questions.add_handler (questions.make_slot ()));
 
-  add_object (heap);
+  this->add_object (heap);
 }
 
 template<typename HeapType>
diff -ru ekiga-3.2.7_orig/lib/engine/presence/heap-impl.h ekiga-3.2.7/lib/engine/presence/heap-impl.h
--- ekiga-3.2.7_orig/lib/engine/presence/heap-impl.h	2012-08-08 19:17:09.767376195 +0200
+++ ekiga-3.2.7/lib/engine/presence/heap-impl.h	2012-08-08 19:19:22.893711886 +0200
@@ -160,7 +160,7 @@
 {
   presentity->questions.add_handler (questions.make_slot ());
 
-  add_object (presentity);
+  this->add_object (presentity);
 }
 
 template<typename PresentityType>
