--- src/tracker-extract/tracker-extract-gif.c~	2012-02-15 15:19:19.000000000 +0000
+++ src/tracker-extract/tracker-extract-gif.c	2013-02-15 02:33:00.584100863 +0000
@@ -617,7 +617,11 @@
 		return FALSE;
 	}	
 
+#ifdef GIFLIB_MAJOR >= 5
+	if ((gifFile = DGifOpenFileHandle (fd, NULL)) == NULL) {
+#else
 	if ((gifFile = DGifOpenFileHandle (fd)) == NULL) {
+#endif
 		PrintGifError ();
 		close (fd);
 		return FALSE;
