diff -Nru java.orig/CyrusSasl/Makefile.in java/CyrusSasl/Makefile.in
--- java.orig/CyrusSasl/Makefile.in	2015-01-12 02:45:30.598298837 +0000
+++ java/CyrusSasl/Makefile.in	2015-01-12 02:46:31.423048432 +0000
@@ -213,7 +213,7 @@
     n|no|NO) false;; \
     *) (install-info --version) >/dev/null 2>&1;; \
   esac
-CLASSPATH_ENV = CLASSPATH=$(JAVAROOT):$(srcdir)/$(JAVAROOT)$${CLASSPATH:+":$$CLASSPATH"}
+CLASSPATH_ENV = CLASSPATH=$(JAVAROOT):$(srcdir)/$(JAVAROOT):$(builddir)/$(JAVAROOT)$${CLASSPATH:+":$$CLASSPATH"}
 am__java_sources = $(javasasl_JAVA)
 am__tagged_files = $(HEADERS) $(SOURCES) $(TAGS_FILES) $(LISP)
 # Read a list of newline-separated strings from the standard input,
