https://bugs.gentoo.org/361461

--- a/chrome/browser/search_engines/template_url_prepopulate_data.h
+++ b/chrome/browser/search_engines/template_url_prepopulate_data.h
@@ -7,6 +7,7 @@
 #pragma once
 
 #include <vector>
+#include <cstddef>
 
 class GURL;
 class PrefService;
--- a/gpu/command_buffer/common/types.h
+++ b/gpu/command_buffer/common/types.h
@@ -11,6 +11,7 @@
 #include <stdint.h>
 #endif
 #include <string>
+#include <cstddef>
 
 typedef signed char         schar;
 typedef signed char         int8;
--- a/third_party/tcmalloc/chromium/src/base/stl_allocator.h
+++ b/third_party/tcmalloc/chromium/src/base/stl_allocator.h
@@ -38,6 +38,7 @@
 #include <config.h>
 
 #include <limits>
+#include <cstddef>
 
 #include "base/basictypes.h"
 #include "base/logging.h"
--- a/third_party/tcmalloc/chromium/src/base/vdso_support.h
+++ b/third_party/tcmalloc/chromium/src/base/vdso_support.h
@@ -37,6 +37,7 @@
 
 #define HAVE_VDSO_SUPPORT 1
 
+#include <cstddef>
 #include <stdlib.h>     // for NULL
 #include <link.h>  // for ElfW
 #include "base/basictypes.h"
--- a/third_party/tcmalloc/chromium/src/common.cc
+++ b/third_party/tcmalloc/chromium/src/common.cc
@@ -34,6 +34,7 @@
 #include "system-alloc.h"
 #include "config.h"
 #include "common.h"
+#include <cstddef>
 
 namespace tcmalloc {
 
--- a/third_party/tcmalloc/chromium/src/symbolize.h
+++ b/third_party/tcmalloc/chromium/src/symbolize.h
@@ -38,6 +38,7 @@
 #include <stdint.h>  // for uintptr_t
 #endif
 #include <map>
+#include <cstddef>
 
 using std::map;
 
--- a/third_party/tcmalloc/chromium/src/system-alloc.h
+++ b/third_party/tcmalloc/chromium/src/system-alloc.h
@@ -37,6 +37,7 @@
 #define TCMALLOC_SYSTEM_ALLOC_H_
 
 #include <config.h>
+#include <cstddef>
 #include "internal_logging.h"
 
 // REQUIRES: "alignment" is a power of two or "0" to indicate default alignment
--- a/ui/gfx/codec/jpeg_codec.h
+++ b/ui/gfx/codec/jpeg_codec.h
@@ -7,6 +7,7 @@
 #pragma once
 
 #include <vector>
+#include <cstddef>
 
 class SkBitmap;
 
--- a/third_party/ots/src/ots.h~	2011-04-04 09:21:06.000000000 +0100
+++ b/third_party/ots/src/ots.h	2011-04-08 14:16:45.000000000 +0100
@@ -9,6 +9,7 @@
 #include <cstdio>
 #include <cstdlib>
 #include <cstring>
+#include <cstddef>
 
 #include "opentype-sanitiser.h"
 
