diff -Nru opal-3.6.7.old/configure.ac opal-3.6.7/configure.ac
--- opal-3.6.7.old/configure.ac	2010-03-09 16:49:38.000000000 +0000
+++ opal-3.6.7/configure.ac	2010-03-09 18:15:32.000000000 +0000
@@ -88,6 +88,7 @@
 OPAL_ZRTP=no
 OPAL_SHARED_LIB=yes
 OPAL_JAVA=yes
+OPAL_ILBC=yes
 OPAL_HAS_MSRP=no
 OPAL_HAS_SIPIM=yes
 OPAL_HAS_RFC4103=yes
@@ -519,6 +520,8 @@
 dnl MSWIN_DEFINE  g711plc,OPAL_G711PLC
 OPAL_SIMPLE_OPTION([g711plc],[OPAL_G711PLC], [whether to enable Packet Loss Concealment for G.711])
 
+OPAL_SIMPLE_OPTION([ilbc],[OPAL_ILBC], [whether to enable the non-Free iLBC plugin])
+
 dnl MSWIN_DISPLAY statistics,Statistics gathering support
 dnl MSWIN_DEFINE  statistics,OPAL_STATISTICS
 OPAL_SIMPLE_OPTION([statistics],[OPAL_STATISTICS], [whether to enable statistics gathering support])
diff -Nru opal-3.6.7.old/Makefile.in opal-3.6.7/Makefile.in
--- opal-3.6.7.old/Makefile.in	2010-02-01 00:19:53.000000000 +0000
+++ opal-3.6.7/Makefile.in	2010-03-09 20:26:19.000000000 +0000
@@ -361,7 +361,6 @@
            $(OPAL_SRCDIR)/codec/g729mf.cxx \
            $(OPAL_SRCDIR)/codec/gsm0610mf.cxx \
            $(OPAL_SRCDIR)/codec/gsmamrmf.cxx \
-           $(OPAL_SRCDIR)/codec/iLBCmf.cxx \
            $(OPAL_SRCDIR)/codec/t38mf.cxx \
            $(OPAL_SRCDIR)/codec/rfc2833.cxx \
            $(OPAL_SRCDIR)/codec/opalwavfile.cxx \
@@ -403,6 +402,10 @@
 
 endif # ifeq ($(OPAL_AEC), yes)
 
+ifeq ($(OPAL_ILBC), yes)
+SOURCES += $(OPAL_SRCDIR)/codec/iLBCmf.cxx
+endif
+
 ####################################################
 
 ifeq ($(OSTYPE),Darwin)
diff -Nru opal-3.6.7.old/plugins/configure.ac opal-3.6.7/plugins/configure.ac
--- opal-3.6.7.old/plugins/configure.ac	2010-03-09 16:49:38.000000000 +0000
+++ opal-3.6.7/plugins/configure.ac	2010-03-09 20:43:15.000000000 +0000
@@ -184,7 +184,18 @@
 dnl     ########################
 dnl     iLBC (system/local)
 dnl     ########################
-OPAL_DETERMINE_ILBC()
+AC_ARG_ENABLE([ilbc],
+              [AC_HELP_STRING([--enable-ilbc],[enable use of non-Free iLBC plugin])],
+              [has_ilbc=$enableval],
+              [has_ilbc=])
+
+if test "$has_ilbc" = "yes" ; then
+   HAVE_ILBC=yes
+   AC_DEFINE(HAVE_ILBC, [1], [Enable iLBC codec])
+   AC_MSG_NOTICE(Enabling iLBC codec)
+   OPAL_DETERMINE_ILBC()
+fi
+AC_SUBST(HAVE_ILBC)
 AC_SUBST(ILBC_SYSTEM)
 AC_SUBST(ILBC_CFLAGS)
 AC_SUBST(ILBC_LIBS)
@@ -543,6 +554,11 @@
 else
 echo "                              SBC :  no"
 fi
+if test "x${HAVE_ILBC}" = "xyes" ; then
+echo "                              iLBC :  yes"
+else
+echo "                              ILBC :  no"
+fi
 
 echo ""
 
diff -Nru opal-3.6.7.old/plugins/Makefile.in opal-3.6.7/plugins/Makefile.in
--- opal-3.6.7.old/plugins/Makefile.in	2010-02-01 00:19:03.000000000 +0000
+++ opal-3.6.7/plugins/Makefile.in	2010-03-09 20:36:53.000000000 +0000
@@ -1,7 +1,6 @@
 SUBDIRS = audio/G722 \
           audio/GSM0610 \
           audio/Speex \
-          audio/iLBC \
 	  audio/LPC_10 \
 	  audio/G726 \
 	  audio/IMA_ADPCM \
@@ -13,6 +12,11 @@
 SUBDIRS += audio/SBC
 endif
 
+HAVE_ILBC=@HAVE_ILBC@
+ifeq ($(HAVE_ILBC),yes)
+SUBDIRS += audio/iLBC
+endif
+
 HAVE_CELT=@HAVE_CELT@
 ifeq ($(HAVE_CELT),yes)
 SUBDIRS += audio/celt
