diff -Nru source.old/common/x86/blockcopy8.asm source/common/x86/blockcopy8.asm
--- source.old/common/x86/blockcopy8.asm	2014-08-21 22:49:02.000000000 +0100
+++ source/common/x86/blockcopy8.asm	2014-11-04 03:19:58.431508738 +0000
@@ -4387,3 +4387,13 @@
     psadbw      xm0, xm4
     movd        eax, xm0
     RET
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/const-a.asm source/common/x86/const-a.asm
--- source.old/common/x86/const-a.asm	2014-08-21 22:49:02.000000000 +0100
+++ source/common/x86/const-a.asm	2014-11-04 03:18:34.242355585 +0000
@@ -104,3 +104,13 @@
 %endrep
 
 const sw_64,       dd 64
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/cpu-a.asm source/common/x86/cpu-a.asm
--- source.old/common/x86/cpu-a.asm	2014-08-21 22:49:02.000000000 +0100
+++ source/common/x86/cpu-a.asm	2014-11-04 03:18:45.546510424 +0000
@@ -195,3 +195,13 @@
     pop r1
     pop r0
     ret
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/dct8.asm source/common/x86/dct8.asm
--- source.old/common/x86/dct8.asm	2014-08-21 22:49:02.000000000 +0100
+++ source/common/x86/dct8.asm	2014-11-04 03:20:15.071736655 +0000
@@ -996,3 +996,13 @@
     RET
 
 %endif ; !HIGH_BIT_DEPTH
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/intrapred16.asm source/common/x86/intrapred16.asm
--- source.old/common/x86/intrapred16.asm	2014-08-21 22:49:02.000000000 +0100
+++ source/common/x86/intrapred16.asm	2014-11-04 03:20:37.208039846 +0000
@@ -12778,3 +12778,13 @@
     dec    r4
     jnz    .loop
     RET
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/ipfilter16.asm source/common/x86/ipfilter16.asm
--- source.old/common/x86/ipfilter16.asm	2014-08-21 22:49:02.000000000 +0100
+++ source/common/x86/ipfilter16.asm	2014-11-04 03:20:46.672169469 +0000
@@ -2892,3 +2892,13 @@
     jnz         .loopH
 
     RET
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/mc-a2.asm source/common/x86/mc-a2.asm
--- source.old/common/x86/mc-a2.asm	2014-08-21 22:49:02.000000000 +0100
+++ source/common/x86/mc-a2.asm	2014-11-04 03:19:33.687169818 +0000
@@ -1131,3 +1131,13 @@
 MBTREE_AVX
 INIT_YMM avx2,fma3
 MBTREE_AVX
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/mc-a.asm source/common/x86/mc-a.asm
--- source.old/common/x86/mc-a.asm	2014-08-21 22:49:02.000000000 +0100
+++ source/common/x86/mc-a.asm	2014-11-04 03:19:26.511071527 +0000
@@ -3720,3 +3720,13 @@
     prefetcht0  [r0+r1*2]
     prefetcht0  [r0+r2]
     RET
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/pixel-a.asm source/common/x86/pixel-a.asm
--- source.old/common/x86/pixel-a.asm	2014-08-21 22:49:02.000000000 +0100
+++ source/common/x86/pixel-a.asm	2014-11-04 03:05:08.691349986 +0000
@@ -6579,3 +6579,13 @@
     mov         [r2], r3w
 .end:
     RET
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/pixeladd8.asm source/common/x86/pixeladd8.asm
--- source.old/common/x86/pixeladd8.asm	2014-08-21 22:49:02.000000000 +0100
+++ source/common/x86/pixeladd8.asm	2014-11-04 03:20:06.287616341 +0000
@@ -738,3 +738,13 @@
 %endmacro
 
 PIXEL_ADD_PS_W64_H2 64, 64
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/pixel-util8.asm source/common/x86/pixel-util8.asm
--- source.old/common/x86/pixel-util8.asm	2014-08-21 22:49:02.000000000 +0100
+++ source/common/x86/pixel-util8.asm	2014-11-04 03:19:45.839336265 +0000
@@ -3697,3 +3697,12 @@
     RET
 %endmacro
 
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/sad16-a.asm source/common/x86/sad16-a.asm
--- source.old/common/x86/sad16-a.asm	2014-08-21 22:49:02.000000000 +0100
+++ source/common/x86/sad16-a.asm	2014-11-04 03:20:47.928186673 +0000
@@ -831,3 +831,12 @@
 SAD_X 4, 16, 16
 SAD_X 4, 16,  8
 
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/ssd-a.asm source/common/x86/ssd-a.asm
--- source.old/common/x86/ssd-a.asm	2014-08-21 22:49:02.000000000 +0100
+++ source/common/x86/ssd-a.asm	2014-11-04 03:18:54.450632387 +0000
@@ -2594,3 +2594,13 @@
     HADDD   m0, m1
     movd    eax, xm0
     RET
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/pixel-32.asm source/common/x86/pixel-32.asm
--- source.old/common/x86/pixel-32.asm	2014-08-21 22:49:02.000000000 +0100
+++ source/common/x86/pixel-32.asm	2014-11-04 03:33:05.798290488 +0000
@@ -418,3 +418,12 @@
     emms
     RET
 
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
