diff -Nru source.old/common/x86/blockcopy8.asm source/common/x86/blockcopy8.asm
--- source.old/common/x86/blockcopy8.asm	2015-10-08 11:01:36.000000000 +0100
+++ source/common/x86/blockcopy8.asm	2015-12-23 06:50:25.398293325 +0000
@@ -5876,3 +5876,13 @@
     dec         r3d
     jnz        .loop
     RET
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/const-a.asm source/common/x86/const-a.asm
--- source.old/common/x86/const-a.asm	2015-10-08 11:01:36.000000000 +0100
+++ source/common/x86/const-a.asm	2015-12-23 06:51:10.697532383 +0000
@@ -139,3 +139,13 @@
 db ((x>>0)&1)+((x>>1)&1)+((x>>2)&1)+((x>>3)&1)+((x>>4)&1)+((x>>5)&1)+((x>>6)&1)+((x>>7)&1)
 %assign x x+1
 %endrep
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/cpu-a.asm source/common/x86/cpu-a.asm
--- source.old/common/x86/cpu-a.asm	2015-10-08 11:01:36.000000000 +0100
+++ source/common/x86/cpu-a.asm	2015-12-23 06:50:25.410293123 +0000
@@ -195,3 +195,13 @@
     pop r1
     pop r0
     ret
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/dct8.asm source/common/x86/dct8.asm
--- source.old/common/x86/dct8.asm	2015-10-08 11:01:36.000000000 +0100
+++ source/common/x86/dct8.asm	2015-12-23 06:51:40.825026321 +0000
@@ -3717,3 +3717,13 @@
     movhps          [r1 + r3], xm1
     RET
 %endif
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/intrapred16.asm source/common/x86/intrapred16.asm
--- source.old/common/x86/intrapred16.asm	2015-10-08 11:01:36.000000000 +0100
+++ source/common/x86/intrapred16.asm	2015-12-23 06:50:25.422292922 +0000
@@ -22069,3 +22069,13 @@
     mov             [r1 + 16], r2w              ; topLast
     mov             [r1 + 32], r3w              ; LeftLast
     RET
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/ipfilter16.asm source/common/x86/ipfilter16.asm
--- source.old/common/x86/ipfilter16.asm	2015-10-08 11:01:36.000000000 +0100
+++ source/common/x86/ipfilter16.asm	2015-12-23 06:52:12.160499988 +0000
@@ -13005,3 +13005,13 @@
 FILTER_VER_CHROMA_AVX2_8x12 ps, 0, 2
 FILTER_VER_CHROMA_AVX2_8x12 sp, 1, 10
 FILTER_VER_CHROMA_AVX2_8x12 ss, 0, 6
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/mc-a2.asm source/common/x86/mc-a2.asm
--- source.old/common/x86/mc-a2.asm	2015-10-08 11:01:36.000000000 +0100
+++ source/common/x86/mc-a2.asm	2015-12-23 06:50:25.438292653 +0000
@@ -1135,3 +1135,13 @@
 MBTREE_AVX
 INIT_YMM avx2,fma3
 MBTREE_AVX
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/mc-a.asm source/common/x86/mc-a.asm
--- source.old/common/x86/mc-a.asm	2015-10-08 11:01:36.000000000 +0100
+++ source/common/x86/mc-a.asm	2015-12-23 06:50:25.442292586 +0000
@@ -5723,3 +5723,13 @@
     prefetcht0  [r0+r1*2]
     prefetcht0  [r0+r2]
     RET
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/pixel-32.asm source/common/x86/pixel-32.asm
--- source.old/common/x86/pixel-32.asm	2015-10-08 11:01:36.000000000 +0100
+++ source/common/x86/pixel-32.asm	2015-12-23 06:50:25.478291981 +0000
@@ -418,3 +418,12 @@
     emms
     RET
 
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/pixel-a.asm source/common/x86/pixel-a.asm
--- source.old/common/x86/pixel-a.asm	2015-10-08 11:01:36.000000000 +0100
+++ source/common/x86/pixel-a.asm	2015-12-23 06:52:41.188012436 +0000
@@ -12186,3 +12186,13 @@
     movd            eax, xm6
     RET
 %endif ; ARCH_X86_64 == 1 && HIGH_BIT_DEPTH == 1
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/pixeladd8.asm source/common/x86/pixeladd8.asm
--- source.old/common/x86/pixeladd8.asm	2015-10-08 11:01:36.000000000 +0100
+++ source/common/x86/pixeladd8.asm	2015-12-23 06:53:10.323523089 +0000
@@ -1144,3 +1144,13 @@
     RET
 
 %endif
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/pixel-util8.asm source/common/x86/pixel-util8.asm
--- source.old/common/x86/pixel-util8.asm	2015-10-08 11:01:36.000000000 +0100
+++ source/common/x86/pixel-util8.asm	2015-12-23 06:53:34.511116856 +0000
@@ -7338,3 +7338,13 @@
     or          eax, r4d
     RET
 %endif ; ARCH_X86_64
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/sad16-a.asm source/common/x86/sad16-a.asm
--- source.old/common/x86/sad16-a.asm	2015-10-08 11:01:36.000000000 +0100
+++ source/common/x86/sad16-a.asm	2015-12-23 06:50:25.470292115 +0000
@@ -1589,3 +1589,12 @@
 SAD_X 4, 64, 48
 SAD_X 4, 64, 64
 
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/ssd-a.asm source/common/x86/ssd-a.asm
--- source.old/common/x86/ssd-a.asm	2015-10-08 11:01:36.000000000 +0100
+++ source/common/x86/ssd-a.asm	2015-12-23 06:50:25.474292048 +0000
@@ -2808,3 +2808,13 @@
     HADDD   m0, m1
     movd    eax, xm0
     RET
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/intrapred8_allangs.asm source/common/x86/intrapred8_allangs.asm
--- source.old/common/x86/intrapred8_allangs.asm	2015-10-08 11:01:36.000000000 +0100
+++ source/common/x86/intrapred8_allangs.asm	2015-12-23 07:08:50.831764329 +0000
@@ -24164,3 +24164,13 @@
     movd        [r0 + 524],     m7              ;byte[5, 6, 7, 8]
 
 RET
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/intrapred8.asm source/common/x86/intrapred8.asm
--- source.old/common/x86/intrapred8.asm	2015-10-08 11:01:36.000000000 +0100
+++ source/common/x86/intrapred8.asm	2015-12-23 07:08:39.039961464 +0000
@@ -18370,3 +18370,13 @@
     mov             [r1 +  8], r2b              ; topLast
     mov             [r1 + 16], r3b              ; LeftLast
     RET
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/ipfilter8.asm source/common/x86/ipfilter8.asm
--- source.old/common/x86/ipfilter8.asm	2015-10-08 11:01:36.000000000 +0100
+++ source/common/x86/ipfilter8.asm	2015-12-23 07:09:08.431470097 +0000
@@ -27834,3 +27834,13 @@
     lea               r0,                [r0 + r1 * 4]
 %endrep
     RET
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/loopfilter.asm source/common/x86/loopfilter.asm
--- source.old/common/x86/loopfilter.asm	2015-10-08 11:01:36.000000000 +0100
+++ source/common/x86/loopfilter.asm	2015-12-23 07:09:16.795330268 +0000
@@ -2279,3 +2279,13 @@
     add         [r1 + 4 * 4], r6d
     RET
 %endif ; ARCH_X86_64
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
diff -Nru source.old/common/x86/sad-a.asm source/common/x86/sad-a.asm
--- source.old/common/x86/sad-a.asm	2015-10-08 11:01:36.000000000 +0100
+++ source/common/x86/sad-a.asm	2015-12-23 07:08:20.660268737 +0000
@@ -4571,3 +4571,13 @@
     RET
 
 %endif
+
+%ifidn __OUTPUT_FORMAT__,elf
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf32
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
+%ifidn __OUTPUT_FORMAT__,elf64
+    section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
