--- ./src/2geom/utils.h~	2010-07-13 04:48:40.000000000 +0100
+++ ./src/2geom/utils.h	2011-04-11 01:23:11.000000000 +0100
@@ -33,6 +33,7 @@
  *
  */
 
+#include <cstddef>
 #include <cmath>
 #include <vector>
 
--- ./src/Makefile.am~	2011-04-11 00:55:41.000000000 +0100
+++ ./src/Makefile.am	2011-04-11 01:32:23.000000000 +0100
@@ -205,7 +205,7 @@
 
 inkscape_SOURCES += main.cpp $(win32_sources)
 inkscape_LDADD = $(all_libs)
-inkscape_LDFLAGS = --export-dynamic $(kdeldflags) $(mwindows)
+inkscape_LDFLAGS = -Wl,--export-dynamic $(kdeldflags) $(mwindows)
 
 inkview_SOURCES += inkview.cpp $(win32_sources)
 inkview_LDADD = $(all_libs)
--- ./src/libavoid/geomtypes.h~	2010-07-13 04:48:40.000000000 +0100
+++ ./src/libavoid/geomtypes.h	2011-04-11 01:08:04.000000000 +0100
@@ -29,6 +29,7 @@
 #ifndef AVOID_GEOMTYPES_H
 #define AVOID_GEOMTYPES_H
 
+#include <cstddef>
 #include <vector>
 #include <utility>
 
