--- gtk/controller/Makefile.am~	2013-01-05 18:36:30.418978195 +0000
+++ gtk/controller/Makefile.am	2013-01-05 19:04:47.054235352 +0000
@@ -81,7 +81,8 @@
 		echo "" ; \
 		exit 1 ; \
 	fi
-	$(VALA_V)$(VALAC) $(VALAFLAGS) $(AM_VALAFLAGS) $(libspice_controller_la_VALASOURCES) -H spice-controller.h
+	$(VALA_V)$(VALAC) $(VALAFLAGS) $(AM_VALAFLAGS) \
+	  $(addprefix $(srcdir)/,$(libspice_controller_la_VALASOURCES)) -H spice-controller.h
 	@touch $@
 
 $(libspice_controller_la_BUILT_SOURCES): controller.vala.stamp
