diff -Nru libpqxx-2.6.8.old/src/binarystring.cxx libpqxx-2.6.8/src/binarystring.cxx
--- libpqxx-2.6.8.old/src/binarystring.cxx	2006-03-31 04:28:22.000000000 +0100
+++ libpqxx-2.6.8/src/binarystring.cxx	2008-11-29 11:25:03.000000000 +0000
@@ -18,6 +18,7 @@
 
 #include <new>
 #include <stdexcept>
+#include <cstring>
 
 #include "libpq-fe.h"
 
diff -Nru libpqxx-2.6.8.old/src/connection_base.cxx libpqxx-2.6.8/src/connection_base.cxx
--- libpqxx-2.6.8.old/src/connection_base.cxx	2006-09-04 12:36:58.000000000 +0100
+++ libpqxx-2.6.8/src/connection_base.cxx	2008-11-29 11:25:21.000000000 +0000
@@ -22,6 +22,7 @@
 #include <cerrno>
 #include <cstdio>
 #include <ctime>
+#include <cstring>
 #include <stdexcept>
 
 #ifdef PQXX_HAVE_SYS_SELECT_H
diff -Nru libpqxx-2.6.8.old/src/cursor.cxx libpqxx-2.6.8/src/cursor.cxx
--- libpqxx-2.6.8.old/src/cursor.cxx	2006-09-04 12:36:58.000000000 +0100
+++ libpqxx-2.6.8/src/cursor.cxx	2008-11-29 11:25:41.000000000 +0000
@@ -18,6 +18,7 @@
 #include "pqxx/compiler-internal.hxx"
 
 #include <cstdlib>
+#include <cstring>
 
 #include "pqxx/cursor"
 #include "pqxx/result"
diff -Nru libpqxx-2.6.8.old/src/result.cxx libpqxx-2.6.8/src/result.cxx
--- libpqxx-2.6.8.old/src/result.cxx	2006-08-29 15:18:37.000000000 +0100
+++ libpqxx-2.6.8/src/result.cxx	2008-11-29 11:26:16.000000000 +0000
@@ -18,6 +18,8 @@
 #include "pqxx/compiler-internal.hxx"
 
 #include <stdexcept>
+#include <cstdlib>
+#include <cstring>
 
 #include "libpq-fe.h"
 
diff -Nru libpqxx-2.6.8.old/src/transaction_base.cxx libpqxx-2.6.8/src/transaction_base.cxx
--- libpqxx-2.6.8.old/src/transaction_base.cxx	2006-09-04 12:36:58.000000000 +0100
+++ libpqxx-2.6.8/src/transaction_base.cxx	2008-11-29 11:26:34.000000000 +0000
@@ -19,6 +19,7 @@
 #include "pqxx/compiler-internal.hxx"
 
 #include <stdexcept>
+#include <cstring>
 
 #include "pqxx/connection_base"
 #include "pqxx/result"
