--- python/virtualenv/virtualenv.py.old	2015-01-26 16:24:28.223351536 +0000
+++ python/virtualenv/virtualenv.py	2015-01-26 16:36:08.251930029 +0000
@@ -22,6 +22,7 @@
 import struct
 import subprocess
 import tarfile
+import xattr
 
 if sys.version_info < (2, 6):
     print('ERROR: %s' % sys.exc_info()[1])
@@ -453,6 +454,12 @@
         shutil.copytree(src, dest, symlink)
     else:
         shutil.copy2(src, dest)
+        try:
+            for attr in xattr.list(src):
+                logger.info('Copying %s from %s to %s',attr,src,dest)
+                xattr.set(dest,attr,xattr.get(src,attr))
+        except (EnvironmentError):
+            logger.info('Failed to copy attributes to %s', dest)
 
 def copyfile(src, dest, symlink=True):
     if not os.path.exists(src):
@@ -1274,6 +1281,12 @@
         ## FIXME: could I just hard link?
         executable = sys.executable
         shutil.copyfile(executable, py_executable)
+        try:
+            for attr in xattr.list(executable):
+                logger.info('Copying %s from %s to %s',attr,executable,py_executable)
+                xattr.set(py_executable,attr,xattr.get(executable,attr))
+        except (EnvironmentError):
+            logger.info('Failed to copy attributes to %s', py_executable)
         make_exe(py_executable)
         if is_win or is_cygwin:
             pythonw = os.path.join(os.path.dirname(sys.executable), 'pythonw.exe')
